components/authz-service/engine/opa/opa.go

```diff
@@ -426,7 +438,8 @@ func (s *State) RulesForProject(
 
 // ListProjectMappings returns a map of all the rules for each projectID.
 func (s *State) ListProjectMappings(ctx context.Context) (map[string][]engine.Rule, error) {
-	rs, err := s.evalQuery(ctx, s.queries[listProjectMapQuery], map[string]interface{}{}, s.v2Store)
+	// NB: V2.1 only, so s.v2p1Store used here
+	rs, err := s.evalQuery(ctx, s.queries[listProjectMapQuery], map[string]interface{}{}, s.v2p1Store)
```

Comment:
:x: this is why tests fail: you've switched `v2p1Store` for `v2Store`. `ingest-service` **still doesn't know whether A2 is running in IAM v2, v1, or v2.1 mode** -- and queries for this *using this API* regardless. So, it'll only keep working as-is if this uses the v2Store.

How does this ever work? Well, the `policy_refresher` doesn't know about IAM v1. So, on server startup, it'll default to updating the `v2Store` if we haven't upgrade to v2.1.

There is likely another bug here: if the server starts in v2.1 mode, ingestion should fail, as the v2Store has never been initialized.

Reply:
Ah, forgot I changed that one for ListProjectMappings, too. Aarggh. Thanks, @srenatus .